iosHyperskillApp/iosHyperskillApp/Sources/Modules/StudyPlan/Views/StudyPlanView.swift

```diff
@@ -105,45 +105,80 @@ struct StudyPlanView: View {
             .frame(maxWidth: .infinity)
         }
     }
+}
+
+// MARK: - StudyPlanView (ViewAction) -
 
-    private func handleViewAction(_ viewAction: StudyPlanScreenFeatureActionViewAction) {
+private extension StudyPlanView {
+    func handleViewAction(
+        _ viewAction: StudyPlanScreenFeatureActionViewAction
+    ) {
         switch StudyPlanScreenFeatureActionViewActionKs(viewAction) {
         case .gamificationToolbarViewAction(let gamificationToolbarViewAction):
-            switch GamificationToolbarFeatureActionViewActionKs(gamificationToolbarViewAction.viewAction) {
-            case .showProfileTab:
-                TabBarRouter(tab: .profile).route()
-            }
+            handleGamificationToolbarViewAction(
+                gamificationToolbarViewAction.viewAction
+            )
         case .problemsLimitViewAction:
             break
         case .studyPlanWidgetViewAction(let studyPlanWidgetViewAction):
-            switch StudyPlanWidgetFeatureActionViewActionKs(studyPlanWidgetViewAction.viewAction) {
-            case .showStageImplementUnsupportedModal:
-                let panModal = StageImplementUnsupportedModalViewController(delegate: viewModel)
-                panModalPresenter.presentPanModal(panModal)
-            case .navigateTo(let navigateToViewAction):
-                switch StudyPlanWidgetFeatureActionViewActionNavigateToKs(navigateToViewAction) {
-                case .stageImplement(let navigateToStageImplementationViewAction):
-                    let assembly = StageImplementAssembly(
-                        projectID: navigateToStageImplementationViewAction.projectId,
-                        stageID: navigateToStageImplementationViewAction.stageId
-                    )
-                    stackRouter.pushViewController(assembly.makeModule())
-                case .stepScreen(let navigateToStepScreenViewAction):
-                    let assembly = StepAssembly(stepRoute: navigateToStepScreenViewAction.stepRoute)
-                    stackRouter.pushViewController(assembly.makeModule())
-                case .home:
-                    TabBarRouter(tab: .home).route()
-                case .selectProject(let navigateToSelectProjectViewAction):
-                    let assembly = ProjectSelectionListAssembly(trackID: navigateToSelectProjectViewAction.trackId)
-                    stackRouter.pushViewController(assembly.makeModule())
-                case .selectTrack:
-                    break
-                }
-            }
+            handleStudyPlanWidgetViewAction(
+                studyPlanWidgetViewAction.viewAction
+            )
+        }
+    }
+
+    func handleGamificationToolbarViewAction(
+        _ viewAction: GamificationToolbarFeatureActionViewAction
+    ) {
+        switch GamificationToolbarFeatureActionViewActionKs(viewAction) {
+        case .showProfileTab:
+            TabBarRouter(tab: .profile).route()
+        }
```

Comment:
I think in feature for features that embed into other features we can create some common delegate that handles view actions, to prevent code duplication and to perform navigation in one place.